cmd/disk-cache.go

```diff
@@ -688,14 +689,12 @@ func (c *cacheObjects) gc(ctx context.Context) {
 			}
 			var wg sync.WaitGroup
 			for _, dcache := range c.cache {
-				if dcache.gcCount() == 0 {
-					continue
-				}
 				wg.Add(1)
 				go func(d *diskCache) {
 					defer wg.Done()
-					d.resetGCCounter()
+					<-d.triggerGCCh
```

Comment:
might want to drain the triggerGCCh after purge routine is done. Otherwise, even for low disk usage this will kick off every 30 mins and exit the purge routine after an unnecessary disk usage check.

Reply:
it will simply block here @poornas if nothing is triggered the purge should only proceed if there is an active trigger by caller.

Reply:
every GET/PUT will put a message in this triggerGCCh as long as the disk usage is high. After purge cycle is done, it would trigger the purge unnecessarily every 30 mins, just because there are more messages in the gcTriggerCh. This is why there was a counter here earlier, and it could just be cleared out after the purge run.

Reply:
Yes but atomic variables are not communication purposes @poornas - they are simply counters using that to spawn go-routines is not idiomatic use of Go.

Channels is how we should be synchronizing this appropriately.